tests/integration/ensure-submodules.js

```diff
@@ -0,0 +1,21 @@
+import { describe, it } from "vitest";
+import assert from "assert";
+import * as fs from "fs";
+
+describe("ensure git submodules are present", function () {
+    it("should have functional tests", function () {
+        try {
+            fs.accessSync("tests/6502_65C02_functional_tests/README.md");
+        } catch {
+            assert.fail("Functional tests submodule missing. Ensure git submodules are fetched.");
```

Comment:
fantastic! thanks for this: perhaps add "Try something like `git submdule update --init`" for those who don't remember the incantation.

Reply:
I'll update after merging